tests/test_integrate.py

```diff
@@ -280,6 +312,60 @@ def test_mc_integration(chunksize, limits):
     ).numpy() == pytest.approx(integral3, rel=0.03)
 
 
+def test_mc_vf_integration():
+    # simple examples
+    num_integral = zintegrate.mc_vf_integrate(
```

Comment:
There comes a crucial thing: can this run with a `z.function` wrapped? Because in the PDF, it usually will be.
You can also swap the default integration methods for PDFs and check if the tests still run, do they?

Reply:
I tried wrapped version and faced some issues. The thing is, vf's `run_integration` is not intended to run under tf.function – there are some python control flow instructions (like "if" in `print_iteration` and numpy math ops when computing results from iterations) and tf.Variable creation (which causes "ValueError: tf.function-decorated function tried to create variables on non-first call"). I tried to use tf.py_function, but it did not help (anyway, i think it's not a relevant option in case of graph-like z.function application). I keep on thinking how to handle this issue.

Reply:
Hm, yep. that's a bit of a problem. One thing you can try is to run it in a [`tf.init_scope`](https://www.tensorflow.org/api_docs/python/tf/init_scope) or just make an issue with VF/ask them. That would be good to be solved somehow